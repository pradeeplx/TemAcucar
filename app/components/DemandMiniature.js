import React, { Component } from 'react'
import { View, TouchableOpacity, Platform } from 'react-native'
import truncate from 'truncate'
import { fontFactor } from "../helpers"
import Colors from "../Colors"
import Icon from "./Icon"
import Sentence from "./Sentence"
import TimeAgo from "./TimeAgo"
import UserImage from "./UserImage"

export default class DemandMiniature extends Component {
  handleView() {
    const { demand, onView, admin } = this.props
    onView(demand, admin)
  }

  handleRefuse() {
    const { demand, onRefuse } = this.props
    onRefuse(demand)
  }

  render() {
    const { demand, currentUser, admin, index } = this.props
    const { user, name, description, distance, created_at, state } = demand
    const showClose = (!admin && currentUser.id !== user.id)
    return (
      <TouchableOpacity onPress={this.handleView.bind(this)} style={{
        backgroundColor: Colors.white,
        borderTopWidth: (index === 0 ? 0 : 12),
        borderColor: Colors.lightBeige,
        flex: 1,
        flexDirection: 'row',
      }}>
        <View style={{
          flex: 3,
          flexDirection: 'column',
          alignItems: 'center',
          marginTop: 10,
        }}>
          <UserImage source={{uri: user.image_url}} />
          <Sentence style={{
            color: Colors.gray,
            marginTop: 6,
            fontSize: 10 * fontFactor(),
          }}>
            A { distance > 1 ? `${Math.round(distance)}km` : `${Math.round(distance * 1000)}m` }
          </Sentence>
        </View>
        <View style={{
          flex: 8,
          flexDirection: 'column',
          paddingVertical: 10,
        }}>
          <Sentence style={[{
            color: Colors.blue,
          }, (Platform.OS === 'ios' ? { fontFamily: 'Avenir-Black' } : { fontWeight: 'bold' })]}>
            { truncate(name, 30) }
          </Sentence>
          <View style={{
            flexDirection: 'row',
          }}>
            <Sentence style={{
              color: Colors.black,
              fontSize: 10 * fontFactor(),
            }}>
              { currentUser.id === user.id ? 'VocÃª' : user.first_name } { 'pediu ' }
            </Sentence>
            <TimeAgo time={created_at} lowerCase={true} style={{
              color: Colors.black,
              fontSize: 10 * fontFactor(),
            }} />
          </View>
          <Sentence style={[{
            color: Colors.darkGray,
            fontSize: 10 * fontFactor(),
            marginTop: 10,
          }]}>
            { truncate(description, 110) }
          </Sentence>
        </View>
        <View style={{
          flex: 1,
          alignItems: 'center',
        }}>
          { showClose && <TouchableOpacity onPress={this.handleRefuse.bind(this)}>
            <Icon name="ios-close-outline" style={{
              backgroundColor: 'transparent',
              color: Colors.gray,
              paddingTop: 4,
              paddingBottom: 4,
              paddingHorizontal: 8,
              fontSize: 22 * fontFactor(),
              textAlign: 'center',
            }} />
          </TouchableOpacity> }
          <Icon name="ios-arrow-forward" style={{
            backgroundColor: 'transparent',
            fontSize: 28 * fontFactor(),
            color: (showClose ? Colors.lightGray : Colors.gray),
            marginTop: (showClose ? 18 : 6),
          }} />
        </View>
      </TouchableOpacity>
    )
  }
}
