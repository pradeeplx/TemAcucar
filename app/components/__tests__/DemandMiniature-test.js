import { View } from 'react-native'
import React from 'react'
import DemandMiniature from '../DemandMiniature'
import renderer from 'react-test-renderer'

const props = {
  demand: {
    user: { id: 1 },
    name: 'Foo',
    description: 'Foo Bar',
    distance: 20,
    created_at: 10E4,
    state: 'bar',
  },
  currentUser: 1,
  index: 1,
  admin: true,
}

it('renders correctly', () => {
  const tree = renderer.create(
    <DemandMiniature {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders borderTopWidth 0 when index is 0', () => {
  const tree = renderer.create(
    <DemandMiniature {...props} index={0} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders showClose styles', () => {
  const tree = renderer.create(
    <DemandMiniature {...props} admin={false} currentUser={2} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders 800m instead of 0.8km for distance', () => {
  const tree = renderer.create(
    <DemandMiniature {...props} demand={{ ...props.demand, distance: 0.8 }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders user first_name if user is not the currentUser', () => {
  const tree = renderer.create(
    <DemandMiniature {...props} currentUser={2} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
