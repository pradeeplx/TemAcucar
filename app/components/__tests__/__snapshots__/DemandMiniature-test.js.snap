exports[`test renders 800m instead of 0.8km for distance 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FAF6F1",
      "borderTopWidth": 12,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "marginTop": 10,
      }
    }>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          undefined,
        ]
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#CECED2",
            "fontSize": 15,
            "marginTop": 6,
          },
        ]
      }>
      A 
      800m
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "flexDirection": "column",
        "paddingVertical": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ]
        }>
         
        pediu 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
              },
            ],
          ]
        }>
        há 47 anos
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#807F7F",
              "fontSize": 15,
              "marginTop": 10,
            },
          ],
        ]
      }>
      Foo Bar
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#CECED2",
            "fontSize": 42,
            "marginTop": 6,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;

exports[`test renders borderTopWidth 0 when index is 0 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FAF6F1",
      "borderTopWidth": 0,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "marginTop": 10,
      }
    }>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          undefined,
        ]
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#CECED2",
            "fontSize": 15,
            "marginTop": 6,
          },
        ]
      }>
      A 
      20km
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "flexDirection": "column",
        "paddingVertical": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ]
        }>
         
        pediu 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
              },
            ],
          ]
        }>
        há 47 anos
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#807F7F",
              "fontSize": 15,
              "marginTop": 10,
            },
          ],
        ]
      }>
      Foo Bar
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#CECED2",
            "fontSize": 42,
            "marginTop": 6,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;

exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FAF6F1",
      "borderTopWidth": 12,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "marginTop": 10,
      }
    }>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          undefined,
        ]
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#CECED2",
            "fontSize": 15,
            "marginTop": 6,
          },
        ]
      }>
      A 
      20km
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "flexDirection": "column",
        "paddingVertical": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ]
        }>
         
        pediu 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
              },
            ],
          ]
        }>
        há 47 anos
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#807F7F",
              "fontSize": 15,
              "marginTop": 10,
            },
          ],
        ]
      }>
      Foo Bar
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#CECED2",
            "fontSize": 42,
            "marginTop": 6,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;

exports[`test renders showClose styles 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FAF6F1",
      "borderTopWidth": 12,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "marginTop": 10,
      }
    }>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          undefined,
        ]
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#CECED2",
            "fontSize": 15,
            "marginTop": 6,
          },
        ]
      }>
      A 
      20km
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "flexDirection": "column",
        "paddingVertical": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ]
        }>
         
        pediu 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
              },
            ],
          ]
        }>
        há 47 anos
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#807F7F",
              "fontSize": 15,
              "marginTop": 10,
            },
          ],
        ]
      }>
      Foo Bar
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Ionicons",
              "fontSize": 27,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {
              "backgroundColor": "transparent",
              "color": "#CECED2",
              "fontSize": 33,
              "paddingBottom": 4,
              "paddingHorizontal": 8,
              "paddingTop": 4,
              "textAlign": "center",
            },
          ]
        }>
        
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#EFEDED",
            "fontSize": 42,
            "marginTop": 18,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;

exports[`test renders user first_name if user is not the currentUser 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FAF6F1",
      "borderTopWidth": 12,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "flexDirection": "column",
        "marginTop": 10,
      }
    }>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 27,
            "height": 54,
            "width": 54,
          },
          undefined,
        ]
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#CECED2",
            "fontSize": 15,
            "marginTop": 6,
          },
        ]
      }>
      A 
      20km
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "flexDirection": "column",
        "paddingVertical": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#44B5B1",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
          ],
        ]
      }>
      Foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Object {
              "color": "#000000",
              "fontSize": 15,
            },
          ]
        }>
         
        pediu 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#654C50",
              "fontFamily": "Avenir",
              "fontSize": 18,
            },
            Array [
              Object {
                "fontSize": 13.5,
              },
              Object {
                "color": "#000000",
                "fontSize": 15,
              },
            ],
          ]
        }>
        há 47 anos
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Array [
            Object {
              "color": "#807F7F",
              "fontSize": 15,
              "marginTop": 10,
            },
          ],
        ]
      }>
      Foo Bar
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Ionicons",
            "fontSize": 27,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "backgroundColor": "transparent",
            "color": "#CECED2",
            "fontSize": 42,
            "marginTop": 6,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;
